src/main/groovy/com/ullink/gradle/opencover/OpenCover.groovy

```diff
@@ -80,10 +104,64 @@ class OpenCover extends ConventionTask {
         if (skipAutoProps) commandLineArgs += '-skipautoprops'
         if (hideSkipped) commandLineArgs += '-hideskipped:' + hideSkipped
 
+        commandLineArgs
+    }
+
+    def runSingleOpenCover(ArrayList commandLineArgs) {
         commandLineArgs += ["-target:${getTargetExec()}", "\"-targetargs:${getTargetExecArgs().collect({escapeArg(it)}).join(' ')}\"", "-targetdir:${project.buildDir}"]
         def filters = getTargetAssemblies().collect { "+[${FilenameUtils.getBaseName(project.file(it).name)}]*" }
         commandLineArgs += '-filter:\\"' + filters.join(' ') + '\\"'
         commandLineArgs += "-output:${getCoverageReportPath()}"
+
+        execute(commandLineArgs)
+    }
+
+    def runMultipleOpenCovers(ArrayList commandLineArgs) {
+        def intermediateReportsPath = new File(reportsFolder, "intermediate-results-" + name)
+        intermediateReportsPath.mkdirs()
+
+        def runs = project.nunit.getTestInputAsList(getWhereFilters())
+
+        GParsPool.withPool {
+            runs.eachParallel {
+                def fileName = getRandomFileName()
+                logger.info("Filename generated for the \'$it\' input was \'$fileName\'")
+
+                def execArgs = project.nunit.buildCommandArgs(it, new File(intermediateReportsPath, fileName + ".xml"))
+                commandLineArgs += ["-target:${getTargetExec()}", "\"-targetargs:${execArgs.collect({escapeArg(it)}).join(' ')}\"", "-targetdir:${project.buildDir}"]
+                commandLineArgs += "-output:${new File(intermediateReportsPath, fileName + ".xml")}"
+
+                execute(commandLineArgs)
+            }
+        }
+
+        execute(getMergeCommand(intermediateReportsPath))
+
+        new File(reportsFolder, "Summary.xml").renameTo(new File(reportsFolder, "coverage.xml"))
+        new File(intermediateReportsPath, "").deleteDir()
+    }
+
+    def getRandomFileName() {
+          UUID.randomUUID().toString()
+      }
+
+    def shouldRunInParallel() {
+        if (!project.nunit.parallelForks || getWhereFilters() == null)
+            return false
+        return true
+    }
+
+    def getWhereFilters() {
+        def nunit = project.nunit
+        if (nunit.hasProperty('where')) {
+            def whereCondition = nunit.where
+            if (whereCondition != null)
```

Comment:
is it possible that the nunit.where expression is null? It's when we have just "-where=" ? I think we should add an assert for that saying that we shouldn't have an empty where expression instead of returning null.

Reply:
See comment above :)